pkg/printers/internalversion/printers.go

```diff
@@ -845,6 +850,8 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 			} else {
 				reason = "NotReady"
 			}
+		} else if reason == "Completed" && hasError {
+			reason = "Error"
```

Comment:
* It seems to me that whether the user gets a detailed message generated by L830-L838 above or this generic "Error" reason is order-dependent. If you change the reason on the container in your test to something more specific, it will continue to pass, but will fail if you change the order of the container statuses in the test case. Instead of a boolean, should we capture an `errorReason` string for use here?
* Related to that, I'd like to see tests covering the reasons generated by L830-L838 above actually getting used. Ideally that coverage would already exist, but it doesn't seem to. 
* I'd also like to see regression coverage for the issue I caught in my previous review, i.e. that running status still takes precedence over error status.

Reply:
Thank you review for this. 

> should we capture an errorReason string for use here?

That's make sense. I add change `hasError` to `errorReason`  and add more test cases to cover it.